src/pydap/model.py

```diff
@@ -503,15 +514,81 @@ class DatasetType(StructureType):
         >>> dataset["B"] = BaseType("B")
         >>> dataset["B"].id
         'B'
-
     """
 
     def __setitem__(self, key, item):
-        StructureType.__setitem__(self, key, item)
+        # StructureType.__setitem__(self, key, item)
```

Comment:
This allows to assign/define a variable within a Dataset with a name that is like
```
Dataset['/Group1/Group2/var_name'] = BaseType('var_name')
```